extensions/amp-ad-network-gmossp-impl/0.1/amp-ad-network-gmossp-impl.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpA4A} from '../../amp-a4a/0.1/amp-a4a';
+import {base64UrlDecodeToBytes} from '../../../src/utils/base64';
+import {dev} from '../../../src/log';
+
+/**
+ * Header that will contain Cloudflare generated signature
+ *
+ * @type {string}
+ * @private
+ */
+const AMP_SIGNATURE_HEADER_ = 'X-AmpAdSignature';
+
+/**
+ * This is a minimalistic AmpA4A implementation that primarily gets an Ad
+ * through a source URL and extracts the generated signature
+ * from a HTTP header.  This is then given to A4A to validate against
+ * the cloudflare signing key.
+ */
+export class AmpAdNetworkGmosspImpl extends AmpA4A {
+
+  /** @override */
+  isValidElement() {
+    return this.isAmpAdElement();
+  }
+
+  /** @override */
+  getSigningServiceNames() {
+    return ['google'];
+  }
+
+  /** @override */
+  getAdUrl() {
+    return this.element.getAttribute('src');
```

Comment:
You'll want to do something like this:

https://github.com/cloudflare/amphtml/blob/master/extensions/amp-ad-network-cloudflare-impl/0.1/amp-ad-network-cloudflare-impl.js#L56-L59

to ensure that you are handling ad urls to your network only.